demo/alloy-showcase-webapp/src/test/java/com/liferay/faces/test/alloy/showcase/datatable/DataTableTester.java

```diff
@@ -0,0 +1,694 @@
+	 * WebElement sendKeys (unless the browser is phantomjs, then it would need the following script on line 81).
+	 * https://seleniumhq.github.io/selenium/docs/api/java/org/openqa/selenium/interactions/Actions.html#sendKeys-java.lang.CharSequence...-
+	 */
+	protected void controlOrCommandClick(Browser browser, String xpath) {
+		Actions commandClickBuilder = browser.createActions();
+		Keys controlOrCommandKey = Keys.CONTROL;
+
+		if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("mac")) {
+			controlOrCommandKey = Keys.COMMAND;
+		}
+
+		if ("phantomjs".equals(browser.getName())) {
+			browser.executeScript(
+				"var element = arguments[0]; YUI().use('node-event-simulate', function(Y) { Y.one(element).simulate('click', { metaKey: true }); });",
+				browser.findElementByXpath(xpath));
+		}
+		else {
+			commandClickBuilder.keyDown(controlOrCommandKey).click(browser.findElementByXpath(xpath)).keyUp(
+				controlOrCommandKey);
+
+			Action commandClick = commandClickBuilder.build();
+			commandClick.perform();
+
+			return;
+		}
+
+		return;
+	}
+
+	protected String dataTableContentXpath(String dataContent) {
+		return "(//table[contains(@class,'alloy-data-table')]/tbody/tr/td[contains(.,'" + dataContent + "')])";
+	}
+
+	protected String dataTablePageAndResultsNumberXpath(int pageNumber, String resultsNumber) {
+		return "//div[@class='showcase-example']//ul[@class='pagination']/li/*[contains(.,'Results " + resultsNumber +
+			" of 162 (Page " + pageNumber + " of 17)')]";
+	}
+
+	protected String dataTablePaginationLinkXpath(String whichPaginationActionLink) {
+		return listItemXpath("ul[@class='pagination']", "li") + "/*[contains(@onclick,\"'" + whichPaginationActionLink +
+			"'\")]";
+	}
+
+	protected String getTableRowCustomersXpath(String firstName, String lastName) {
+		return "//tbody/tr[contains(.,'" + firstName + "')][contains(.,'" + lastName +
+			"')]//input[contains(@id,'customers:')]";
+	}
+
+	protected String getTableRowXpath(String firstName, String lastName) {
+		return "//tbody/tr[contains(.,'" + firstName + "')][contains(.,'" + lastName + "')]";
+	}
+
+	protected String labelSelectOptionValueXpath(String labelText, String optionValue) {
+		return "//label[contains(.,'" + labelText + "')]/following-sibling::select/option[@value='" + optionValue +
+			"']";
+	}
+
+	protected void runDataTableLazyTest(String componentName) throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser, componentName, "lazy");
+
+		testPagesPaginationDefault(browser);
+	}
+
+	protected void runDataTablePaginationTest(String componentName) throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser, componentName, "pagination");
+
+		testPagesPaginationDefault(browser);
+
+		// Summary Position
+		// Test default "bottom"
+		SeleniumAssert.assertElementVisible(browser,
+			labelSelectOptionValueXpath("Summary Position", "bottom'][@selected='selected"));
```

Comment:
As we discussed on our call, passing and xpath fragment with unopened or unclosed brackets or quotes is confusing since it violates [encapsulation](http://stiemannkj1.gitlab.io/encapsulating-functionality/). In other words, the caller must know about the internals of method to even pass this value, but the whole point of using methods is to avoid knowing the internals of them. Please refactor any code that uses fragments like `"bottom'][@selected='selected"` to use xpath fragments like `/option[@value='bottom'][@selected='selected']` (where all brackets and quotes are opened and closed in the same `String`).